### ISSUE TITLE
announcements for all people coming together

### ISSUE THREAD

**Issue by sunsgs** [CCOSTAN/Home-AssistantConfig#767]
Hello,
I have seen your announcement when a single person is coming in , did u handle when all are coming together?

thanks

**Comment by CCOSTAN** [comment]
Hi @sunsgs , I don't really handle multiple people well.  I just cycle through them as single announcements.  We all have iOS so the system can detect us at slightly different times.  I could announce all the people that are home but  I wouldn't be able to determine who got home and who was home.  One idea is to extend the presence detection #275 and then you would have a better base on who is arriving and who is here.  and then you could adjust the speech appropriately. 

Hasn't really been a big deal for me so I haven't dived too deep into it yet to implement and answer. 
(if you put something in place, please let me know though! :) )

**Comment by sunsgs** [comment]
Just to let you know what I have achived (i am unable to set a gir repo on HA at the moment)

Btw i've created an empty group on HA startup called occupancy
```
Occupancy:
  name: Occupancy
  entities: []
```

than i have an automation that detects who has arrived home
 ```
 trigger:
    - platform: state
      entity_id:
        - device_tracker.XXX
        - device_tracker.XXXXYYYYY
      from: 'not_home'  
      to: 'home'
```
and runs a script

```
    - wait_template: "{{ is_state('binary_sensor.window_door_sensor', 'on') }}"
      timeout: '00:30:00'
      continue_on_timeout: 'false'
    - service: notify.notify
      data_template:
        title: Welcome back
        message: >
          {% set person = expand('group.occupancy')|map(attribute='name')|join(' and ') %}
          Welcome back {{ person }}
    - service: script.say
      data_template:
        message: >
          {% set count = expand('group.occupancy') | count %}
          {% set person = expand('group.occupancy')|map(attribute='name')|join(' and ') %}
          {% if count == 1 %}
          {% set phrases = ["Hello! " ~ person+" is arrived home "] %}
          {% else %}
          {% set phrases = ["Hello. Nice to see you back home my lords"] %}
          {% endif %}
          {{ phrases | random}}
    - service: group.set
      data_template:
        object_id: "occupancy"
        entities: []
```

Sorry but I Am pretty new in HA

**Comment by CCOSTAN** [comment]
This is GREAT @sunsgs - Can you post the code where you fill in the occupancy group?  I love this and am going to implement it.

**Comment by sunsgs** [comment]
I am just uploading my repo, any tip for gitignore ? i have basically your setup, i have read all your files! :D

**Comment by CCOSTAN** [comment]
> I am just uploading my repo, any tip for gitignore ? i have basically your setup, i have read all your files! :D

You should be good.  Just make sure you are using Secrets.yaml and aren't exposing anything (passwords, location, etc) that you don't want to.  Looking forward to following your repo! Thanks for sharing!

**Comment by sunsgs** [comment]
this is my repo https://github.com/sunsgs/home-assistant

basically you can find all the steps here:

https://github.com/sunsgs/home-assistant/blob/master/groups.yaml
https://github.com/sunsgs/home-assistant/blob/master/automation/greetings.yaml
https://github.com/sunsgs/home-assistant/blob/master/script/say_greeting.yaml

please let me know if u think something can be improved, thanks

**Comment by CCOSTAN** [comment]
Thanks for the nudge @sunsgs . Looks like I have it all implemented to handle multiple arrivals as long as the system picks them up quick. :) 

I made some enhancements to handle is/are in the announcements based on multiple people as well.  

Thanks for the inspiration! 🎉

**Comment by sunsgs** [comment]
cool , I preferred to switch the arrays. In the future i would like to undesterand if someone in the family group or children group has arrived so I can trigger announcements and/or actions.

**Comment by poudenes** [comment]
This seems to work...

automation to create the group.
```
- alias: "TEST - Create arrive group"
  trigger:
    - platform: state
      entity_id:
        - input_boolean.kay_home_manual
        - input_boolean.peter_home_manual
      from: 'off'
      to: 'on'

  action:
    - service: group.set
      data_template:
        object_id: "arriving"
        add_entities: >-
          {{ trigger.to_state.entity_id }}
```

automation to do the actual greetings when sensor turned on.
```
- alias: "Notify - Greets People Home"
  trigger:
    - platform: state
      entity_id:
        - input_boolean.peter_home_manual
        - input_boolean.kay_home_manual
      from: "off"
      to: "on"

  action:
    - wait_template: >-
        {{ (states("binary_sensor.fibaro_imac_custom_sensor") == 'on') }}

    - delay: "00:00:10"

    - service: script.engine_say

     < . rest of the script. >

    - service: group.set
      data_template:
        object_id: "arriving"
        entities: []
```

**Comment by poudenes** [comment]
The script I post work for me now 4 days in a row. When we come home separate or together.
First part only collect the entities and second one will trigger when entity turn on and wait when door goes open.
In meanwhile all entities went on outside and collected in the group.

At least the second script clears the group again

**Comment by sunsgs** [comment]
What was not working? The script posted by me and implemented by ccostan it was working

**Comment by poudenes** [comment]
Not when 2 persons went home. Only 1 person was announced

**Comment by sunsgs** [comment]
I don’t know. My code was working for all people 1 or multiple. That was the sense of the script. In your one I cannot see any difference from mine. That’s why I was asking

**Comment by poudenes** [comment]
I tried using the script but then i only get 1 person that been announced. 
I thought it was because of the wait moment. That automation not start again because its already running and have the wait moment. So second person isn't add to the group.

The way how i do it now is working for me. HA Version 0.112.4. Maybe in newer HA it wont work your script?

**Comment by poudenes** [comment]
> I don’t know. My code was working for all people 1 or multiple. That was the sense of the script. In your one I cannot see any difference from mine. That’s why I was asking

Currently if an automation triggers more than once while waiting in a delay or wait template it will cancel wait templates/delay and move to the next step in the sequence. This has been fixed in 0.113.. So will wait for the release and maybe I can combine them again
